core/src/Kokkos_Core_fwd.hpp

```diff
@@ -330,6 +332,131 @@ struct LAnd;
 template <class ScalarType, class Space = HostSpace>
 struct LOr;
 
+//--------------------------------------------------------------------------------------//
+//  this is used to mark execution or memory spaces as not available
+//
+template <typename Tp>
+struct IsSpaceAvailable : std::true_type {};
+
+//--------------------------------------------------------------------------------------//
+//  declare any spaces that might not be available and mark them as unavailable
+//
+#if !defined(KOKKOS_ENABLE_SERIAL)
+class Serial;
```

Comment:
this Serial still doesn't seem to be moved out of here?

Reply:
Yes, you're right. I want to combine this with the `fwd/Kokkos_FWD*` files anyway in the next commit.